From 6bbf6b001e085025cf33412b15eb79d46e2e670c Mon Sep 17 00:00:00 2001
From: Bruno Pitrus <brunopitrus@hotmail.com>
Date: Tue, 11 Apr 2023 19:32:13 +0000
Subject: [PATCH] Add missing includes causing build errors with GCC/libstdc++
 13

Bug: 957519
Change-Id: If09ae682695714a19e37f3830013a003caba025a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4330267
Reviewed-by: Ken Rockot <rockot@google.com>
Auto-Submit: Bruno Pitrus <brunopitrus@hotmail.com>
Reviewed-by: Reilly Grant <reillyg@chromium.org>
Commit-Queue: Ken Rockot <rockot@google.com>
Reviewed-by: Robert Kaplow <rkaplow@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1128822}
---
 components/metrics/psi_memory_parser.h                         | 1 +
 components/services/app_service/public/cpp/capability_access.h | 1 +
 components/soda/constants.h                                    | 1 +
 device/base/synchronization/one_writer_seqlock.h               | 1 +
 device/bluetooth/public/cpp/bluetooth_uuid.h                   | 1 +
 extensions/common/constants.h                                  | 3 +++
 extensions/renderer/bindings/api_invocation_errors.h           | 1 +
 net/base/parse_number.h                                        | 2 ++
 net/cookies/cookie_inclusion_status.h                          | 1 +
 sandbox/linux/syscall_broker/broker_file_permission.h          | 1 +
 .../blink/public/common/bluetooth/web_bluetooth_device_id.h    | 3 +++
 .../public/common/origin_trials/origin_trial_public_key.h      | 1 +
 ui/gfx/geometry/linear_gradient.h                              | 2 ++
 13 files changed, 19 insertions(+)

diff --git a/components/metrics/psi_memory_parser.h b/components/metrics/psi_memory_parser.h
index b8b22f4ab74..bf02be5074e 100644
--- a/components/metrics/psi_memory_parser.h
+++ b/components/metrics/psi_memory_parser.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_METRICS_PSI_MEMORY_PARSER_H_
 #define COMPONENTS_METRICS_PSI_MEMORY_PARSER_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/gtest_prod_util.h"
diff --git a/components/services/app_service/public/cpp/capability_access.h b/components/services/app_service/public/cpp/capability_access.h
index 991a993ba99..8050a888d81 100644
--- a/components/services/app_service/public/cpp/capability_access.h
+++ b/components/services/app_service/public/cpp/capability_access.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_SERVICES_APP_SERVICE_PUBLIC_CPP_CAPABILITY_ACCESS_H_
 #define COMPONENTS_SERVICES_APP_SERVICE_PUBLIC_CPP_CAPABILITY_ACCESS_H_
 
+#include <memory>
 #include <string>
 #include <utility>
 
diff --git a/components/soda/constants.h b/components/soda/constants.h
index 82b2b98b2fa..b455e2a97da 100644
--- a/components/soda/constants.h
+++ b/components/soda/constants.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_SODA_CONSTANTS_H_
 #define COMPONENTS_SODA_CONSTANTS_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/files/file_path.h"
diff --git a/device/base/synchronization/one_writer_seqlock.h b/device/base/synchronization/one_writer_seqlock.h
index e37a16b81be..b93f14140c6 100644
--- a/device/base/synchronization/one_writer_seqlock.h
+++ b/device/base/synchronization/one_writer_seqlock.h
@@ -6,6 +6,7 @@
 #define DEVICE_BASE_SYNCHRONIZATION_ONE_WRITER_SEQLOCK_H_
 
 #include <atomic>
+#include <cstddef>
 #include <cstdint>
 #include <type_traits>
 
diff --git a/device/bluetooth/public/cpp/bluetooth_uuid.h b/device/bluetooth/public/cpp/bluetooth_uuid.h
index b4bedf01570..d636ba62e2f 100644
--- a/device/bluetooth/public/cpp/bluetooth_uuid.h
+++ b/device/bluetooth/public/cpp/bluetooth_uuid.h
@@ -5,6 +5,7 @@
 #ifndef DEVICE_BLUETOOTH_PUBLIC_CPP_BLUETOOTH_UUID_H_
 #define DEVICE_BLUETOOTH_PUBLIC_CPP_BLUETOOTH_UUID_H_
 
+#include <cstdint>
 #include <ostream>
 #include <string>
 #include <vector>
diff --git a/extensions/common/constants.h b/extensions/common/constants.h
index 0f7db327206..284cc878ff8 100644
--- a/extensions/common/constants.h
+++ b/extensions/common/constants.h
@@ -5,6 +5,9 @@
 #ifndef EXTENSIONS_COMMON_CONSTANTS_H_
 #define EXTENSIONS_COMMON_CONSTANTS_H_
 
+#include <cstddef>
+#include <cstdint>
+
 #include "base/files/file_path.h"
 #include "base/strings/string_piece_forward.h"
 #include "build/chromeos_buildflags.h"
diff --git a/extensions/renderer/bindings/api_invocation_errors.h b/extensions/renderer/bindings/api_invocation_errors.h
index e21700ecfc4..3b704e83b06 100644
--- a/extensions/renderer/bindings/api_invocation_errors.h
+++ b/extensions/renderer/bindings/api_invocation_errors.h
@@ -5,6 +5,7 @@
 #ifndef EXTENSIONS_RENDERER_BINDINGS_API_INVOCATION_ERRORS_H_
 #define EXTENSIONS_RENDERER_BINDINGS_API_INVOCATION_ERRORS_H_
 
+#include <cstdint>
 #include <set>
 #include <string>
 
diff --git a/net/base/parse_number.h b/net/base/parse_number.h
index f4827f7a7da..f923b18a7f4 100644
--- a/net/base/parse_number.h
+++ b/net/base/parse_number.h
@@ -5,6 +5,8 @@
 #ifndef NET_BASE_PARSE_NUMBER_H_
 #define NET_BASE_PARSE_NUMBER_H_
 
+#include <cstdint>
+
 #include "base/strings/string_piece.h"
 #include "net/base/net_export.h"
 
diff --git a/net/cookies/cookie_inclusion_status.h b/net/cookies/cookie_inclusion_status.h
index 51fb970be72..e8dd6d8d9fa 100644
--- a/net/cookies/cookie_inclusion_status.h
+++ b/net/cookies/cookie_inclusion_status.h
@@ -6,6 +6,7 @@
 #define NET_COOKIES_COOKIE_INCLUSION_STATUS_H_
 
 #include <bitset>
+#include <cstdint>
 #include <ostream>
 #include <string>
 #include <vector>
diff --git a/sandbox/linux/syscall_broker/broker_file_permission.h b/sandbox/linux/syscall_broker/broker_file_permission.h
index 3538da53ca9..e8e32cccd60 100644
--- a/sandbox/linux/syscall_broker/broker_file_permission.h
+++ b/sandbox/linux/syscall_broker/broker_file_permission.h
@@ -6,6 +6,7 @@
 #define SANDBOX_LINUX_SYSCALL_BROKER_BROKER_FILE_PERMISSION_H_
 
 #include <bitset>
+#include <cstdint>
 #include <string>
 
 #include "base/strings/string_piece.h"
diff --git a/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h b/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
index 8ab06791464..329c8333cdd 100644
--- a/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
+++ b/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
@@ -8,6 +8,9 @@
 #include <stdint.h>
 
 #include <array>
+#include <cstdint>
+#include <functional>
+#include <iosfwd>
 #include <string>
 
 #include "third_party/blink/public/common/common_export.h"
diff --git a/third_party/blink/public/common/origin_trials/origin_trial_public_key.h b/third_party/blink/public/common/origin_trials/origin_trial_public_key.h
index c1b5f59f57e..246200991a5 100644
--- a/third_party/blink/public/common/origin_trials/origin_trial_public_key.h
+++ b/third_party/blink/public/common/origin_trials/origin_trial_public_key.h
@@ -6,6 +6,7 @@
 #define THIRD_PARTY_BLINK_PUBLIC_COMMON_ORIGIN_TRIALS_ORIGIN_TRIAL_PUBLIC_KEY_H_
 
 #include <array>
+#include <cstdint>
 
 namespace blink {
 
diff --git a/ui/gfx/geometry/linear_gradient.h b/ui/gfx/geometry/linear_gradient.h
index d337a7de29b..d327662833f 100644
--- a/ui/gfx/geometry/linear_gradient.h
+++ b/ui/gfx/geometry/linear_gradient.h
@@ -8,6 +8,8 @@
 #include <stdint.h>
 
 #include <array>
+#include <cstddef>
+#include <cstdint>
 #include <string>
 
 #include "ui/gfx/geometry/geometry_skia_export.h"
